sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -68,7 +68,20 @@ trait StringRegexExpression extends ImplicitCastInputTypes {
  * Simple RegEx pattern matching function
  */
 @ExpressionDescription(
-  usage = "str _FUNC_ pattern - Returns true if str matches pattern and false otherwise.")
+  usage = "str _FUNC_ pattern - Returns true if str matches pattern, " +
+    "null if any arguments are null, false otherwise.",
+  extended =
+    "The pattern is a string which is matched literally, with exception to the following " +
+    "special symbols:\n\n" +
+    "    _ matches any one character in the input (similar to . in posix " +
+    "regular expressions)\n\n" +
+    "    % matches zero ore more characters in the input (similar to .* in " +
+    "posix regular expressions)\n\n" +
+    "The escape character is '\\'. If an escape character precedes a special symbol or " +
+    "another escape character, the following character is matched literally, For example, " +
+    "the expression `<path> like \\%SystemDrive\\%\\\\Users%` will match any `<path>` that " +
+    "starts with '%SystemDrive%\\Users'. It is invalid to escape any other character.\n\n" +
+    "Use RLIKE to match with standard regular expressions.")
```

Comment:
https://github.com/apache/spark/pull/15513 is adding examples and changing the function description format and contents for each function. Could you check it and try to make it consistent with the others?